foyer-storage/src/device/direct_file.rs

```diff
@@ -87,14 +91,21 @@ impl DirectFileDevice {
             capacity = self.capacity,
         );
 
-        let file = self.file.clone();
+        #[allow(unused_mut)]
+        let mut file = self.file.clone();
+
         asyncify_with_runtime(&self.runtime, move || {
-            #[cfg(target_family = "unix")]
-            use std::os::unix::fs::FileExt;
             #[cfg(target_family = "windows")]
-            use std::os::windows::fs::FileExt;
+            let written = {
+                let original = file.stream_position()?;
+                let written = file.seek_write(buf.as_aligned(), offset)?;
+                file.seek(std::io::SeekFrom::Start(original))?;
+                written
+            };
+
+            #[cfg(target_family = "unix")]
+            let written = { file.write_at(buf.as_aligned(), offset)? };
```

Comment:
Hi, maybe there is some misunderstanding. IMO, there is no need to restore the fd seek position after doing positioned io.

Reply:
Yeah just wanted to make sure that we are not relying on write_at's behavior of not moving the cursor. Removed the useless seeking.